### PR TITLE
refactor(adyen): use enums instead of strings for statuses

### LINKED ISSUE #262: refactor(adyen): Use enums instead of strings for statuses
As of now, we use string literals for matching on status codes returned by Adyen. They need to be replaced by enums instead.

3 instances I could find are:

https://github.com/juspay/orca/blob/7348d76cad8f52548f84c6a606fb177e7ca3620e/crates/router/src/connector/adyen/transformers.rs#L431-L435

https://github.com/juspay/orca/blob/7348d76cad8f52548f84c6a606fb177e7ca3620e/crates/router/src/connector/adyen/transformers.rs#L470-L476

https://github.com/juspay/orca/blob/7348d76cad8f52548f84c6a606fb177e7ca3620e/crates/router/src/connector/adyen/transformers.rs#L684-L701

Maybe even this:

https://github.com/juspay/orca/blob/7348d76cad8f52548f84c6a606fb177e7ca3620e/crates/router/src/connector/adyen.rs#L601-L604

Any other similar usages of strings for status codes must be addressed.

If you're interested in picking it up, assign this issue to yourself, or if you don't have access, let me know and I'll assign it to you.

### PR DESCRIPTION
**Pull request by prasunna09**
Fixes #262

## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description
Used enums instead of strings for statuses for Adyen.

## Checklist

- [x] I formatted the code `cargo +nightly fmt`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/adyen/transformers.rs [hunk 1]
```diff
@@ -62,6 +62,29 @@ struct AdyenBrowserInfo {
     java_enabled: bool,
 }
 
+#[derive(Debug, Clone, Deserialize)]
+pub enum AdyenStatus{
+    Authorised,
+    Refused,
+    Cancelled,
+    RedirectShopper,
+
```
Comment (Narayanbhat166) [inline crates/router/src/connector/adyen/transformers.rs:71]: There is an extra line here
run cargo +nightly fmt to format the code

File: crates/router/src/connector/adyen/transformers.rs [hunk 2]
```diff
@@ -455,18 +473,18 @@ pub fn get_adyen_response(
...
                 storage_enums::AttemptStatus::Authorized
             } else {
                 storage_enums::AttemptStatus::Charged
             }
         }
-        "Refused" => storage_enums::AttemptStatus::Failure,
+        AdyenStatus::Refused => storage_enums::AttemptStatus::Failure,
         _ => storage_enums::AttemptStatus::Pending,
     };
+    //let status = response.result_code.into();
```
Comment (SanchithHegde) [inline crates/router/src/connector/adyen/transformers.rs:487]: Remove this comment

**Review by SanchithHegde — CHANGES_REQUESTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — APPROVED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 3]
```diff
@@ -73,6 +73,24 @@ struct AdyenBrowserInfo {
...
+    Refused,
+    Cancelled,
+    RedirectShopper,
+}
+
+impl From<AdyenStatus> for storage_enums::AttemptStatus {
+    fn from(item: AdyenStatus) -> Self {
+        match item {
+            AdyenStatus::Authorised => Self::Charged,
+            AdyenStatus::Refused | AdyenStatus::Cancelled => Self::Failure,
```
Comment (dracarys18) [inline crates/router/src/connector/adyen/transformers.rs:88]: `Cancelled` status should be mapped to `Self::Voided`. Correct me if I am wrong @SanchithHegde .

**Review by dracarys18 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/adyen/transformers.rs (hunk 3 above)
Comment (SanchithHegde) [reply crates/router/src/connector/adyen/transformers.rs:88]: You're right.

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

**Review by jarnura — APPROVED** [review]

**Review by dracarys18 — APPROVED** [review]

**Pull request merged**